rudder-server-relay/SOURCES/rudder-pkg-files/rudderPkgUtils.py

```diff
@@ -0,0 +1,326 @@
+            return False
+      else:
+        has_depends = True
+        if not depends_printed:
+          logging.info("This package depends on the following")
+          depends_printed = True
+        logging.info("  on " + system + " : " + ", ".join(metadata["depends"][system]))
+    if has_depends:
+      logging.info("It is up to you to make sure those dependencies are installed")
+  return True
+
+
+def extract_scripts(metadata,package_file):
+  package_dir = DB_DIRECTORY + "/" + metadata["name"]
+  shell("mkdir -p " + package_dir + "; ar p '" + package_file + "' scripts.txz | tar xJ --no-same-owner -C " + package_dir)
+  return package_dir
+
+
+def run_script(name, script_dir, exist):
+  script = script_dir + "/" + name 
+  if os.path.isfile(script):
+    if exist is None:
+      param = ""
+    elif exist:
+      param = "upgrade"
+    else:
+      param = "install"
+    shell(script + " " + param)
+
+
+def jar_status(name, enable):
+  global jetty_needs_restart
+  text = open(PLUGINS_CONTEXT_XML).read()
+  def repl(match):
+    enabled = [x for x in match.group(1).split(',') if x != name and x != '']
+    pprint(enabled)
+    if enable:
+      enabled.append(name)
+    plugins = ','.join(enabled)
+    return '<Set name="extraClasspath">' + plugins + '</Set>'
+  text = re.sub(r'<Set name="extraClasspath">(.*?)</Set>', repl, text)
+  open(PLUGINS_CONTEXT_XML, "w").write(text)
+  jetty_needs_restart = True
+
+
+def remove_files(metadata):
+  for filename in reversed(metadata['files']):
+    # ignore already removed files
+    if not os.path.exists(filename):
+      logging.info("Skipping removal of " + filename + " as it does not exist")
+      continue
+
+    # remove old files
+    if filename.endswith('/'):
+      try:
+        os.rmdir(filename)
+      except OSError:
+        pass
+    else:
+      os.remove(filename)
+
+
+def install(metadata, package_file, exist):
+  if exist:
+    remove_files(DB['plugins'][metadata['name']])
+  # add new files
+  files = []
+  for tarfile in metadata['content']:
+    dest = metadata['content'][tarfile]
+    (_, file_list, _) = shell("mkdir -p " + dest + "; ar p '" + package_file + "' " + tarfile + " | tar xJv --no-same-owner -C " + dest, keep_output=True)
+    files.append(dest+'/')
+    files.extend([ dest + '/' + x for x in file_list.split("\n") if x != ''])
+
+  metadata['files'] = files
+  # update db
+  DB['plugins'][metadata['name']] = metadata
+  db_save()
+
+############# Variables ############# 
+""" Defining global variables."""
+
+LOG_PATH = "/var/log/rudder/rudder-pkg.log"
+CONFIG_PATH = "/opt/rudder/etc/rudder-pkg.conf"
+FOLDER_PATH = "/var/rudder/tmp/plugins"
+INDEX_PATH = FOLDER_PATH + "/rpkg.index"
+GPG_HOME = "/tmp/toto"
+GPG_RUDDER_KEY = "/var/rudder/rudder_apt_key.pub"
+GPG_RUDDER_KEY_FINGERPRINT = "7C16 9817 7904 212D D58C  B4D1 9322 C330 474A 19E8"
+
+# Repos specific variables
+try:
```

Comment:
We should not fail if the command used does not involve the repos, like rudder-pkg install-file or rudder-pkg list